app/models/P5/departure/HouseConsignment.scala

```diff
@@ -19,13 +19,38 @@ package models.P5.departure
 import play.api.libs.json.Json
 import play.api.libs.json.OFormat
 
-case class HouseConsignment(ConsignmentItem: Seq[ConsignmentItem]) {
+case class HouseConsignment(
+  ConsignmentItem: Seq[ConsignmentItem],
+  Consignor: Option[Consignor],
+  Consignee: Option[Consignee],
+  PreviousDocument: Option[List[PreviousDocument]],
+  TransportDocument: Option[List[TransportDocument]],
+  SupportingDocument: Option[List[SupportingDocument]],
+  AdditionalInformation: Option[List[AdditionalInformation]],
+  AdditionalReference: Option[List[AdditionalReference]],
+  TransportCharges: Option[TransportCharges],
+  grossMass: Double,
+  countryOfDispatch: Option[String],
+  referenceNumberUCR: Option[String]
+) {
 
   val totalPackages: Int = ConsignmentItem.foldLeft(0)(
     (total, item) => total + item.totalPackages
   )
 
   val totalItems: Int = ConsignmentItem.length
+
+  val previousDocumentInHC: Option[String] = PreviousDocument.map(_.map(_.toString)).map(_.mkString("; "))
```

Comment:
Could refactor these to be `.map(_.showAll)` using the helper methods in https://github.com/hmrc/transit-movements-trader-manage-documents/blob/main/app/models/P5/package.scala. Just means if we suddenly decide to change the separator from `; ` to something else we don't need to do so in loads of different places.